src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs

```diff
@@ -594,10 +594,53 @@ public unsafe void Reset()
         {
             _updateStopwatch.Stop();
             _renderStopwatch.Stop();
-            _glfw.DestroyWindow(_windowPtr);
+
+            try
+            {
+                _glfw.DestroyWindow(_windowPtr);
+            }
+#pragma warning disable 168
+            // If the window is already destroyed, it throws an exception,
+            // but we want the window destroyed anyways, so just ignore it
+            catch(GlfwException _e)
+#pragma warning restore 168
+            {
+                
+            }
+            
             _windowPtr = (WindowHandle*) 0;
         }
 
+        private void Dispose(bool disposing)
+        {
+            unsafe
+            {
+                Reset();
+                _glfw.GcUtility.Unpin(_onClosing);

```

Comment:
This may throw an exception when disposing a window that's never been opened, as `_onClosing` (and related callbacks) will be null in that case. Also I'm not sure when `_glfw` is assigned but we should probably allow that being null too, just to make the dispose method as unbreakable as possible.

Reply:
How should I go about avoiding that?

Reply:
Probably null checks, but seems kinda messy.